lib/accessory.js

```diff
@@ -60,6 +87,86 @@ class HygrothermographAccessory {
     return this.latestTemperature + this.temperatureOffset;
   }
 
+  setIlluminance(newValue, force = false) {
+    const MinMaxIlluminanceValues = { min: 0.0001, max: 100000 };
+    if (newValue == null) {
+      return;
+    }
+    // Validate values according to https://developers.homebridge.io/#/characteristic/CurrentAmbientLightLevel
+    newValue =
+      newValue > MinMaxIlluminanceValues.max
+        ? MinMaxIlluminanceValues.max
+        : newValue;
+    newValue =
+      newValue < MinMaxIlluminanceValues.min
+        ? MinMaxIlluminanceValues.min
+        : newValue;
+    this.latestIlluminance = newValue;
+    this.lastUpdatedAt = Date.now();
+    if (this.useBatchUpdating && force === false) {
+      return;
+    }
+    this.illuminanceService
+      .getCharacteristic(Characteristic.CurrentAmbientLightLevel)
+      .updateValue(newValue);
+    this.addFakeGatoHistoryEntry();
+    this.publishValueToMQTT(this.illuminanceMQTTTopic, this.illuminance);
+  }
+
+  get illuminance() {
+    if (this.hasTimedOut() || this.latestIlluminance == null) {
+      return;
+    }
+    return this.latestIlluminance + this.temperatureOffset;
+  }
+
+  setMoisture(newValue, force = false) {
+    if (newValue == null) {
+      return;
+    }
+    this.latestMoisture = newValue;
+    this.lastUpdatedAt = Date.now();
+    if (this.useBatchUpdating && force === false) {
+      return;
+    }
+    this.moistureService
+      .getCharacteristic(Characteristic.CurrentRelativeHumidity)
+      .updateValue(newValue);
+    this.addFakeGatoHistoryEntry();
+    this.publishValueToMQTT(this.moistureMQTTTopic, this.moisture);
+  }
+
+  get moisture() {
+    if (this.hasTimedOut() || this.latestMoisture == null) {
+      return;
+    }
+    return this.latestMoisture + this.moistureOffset;
+  }
+
+  setFertility(newValue, force = false) {
+    if (newValue == null) {
+      return;
+    }
+    this.latestFertility = newValue;
+    this.lastUpdatedAt = Date.now();
+    if (this.useBatchUpdating && force === false) {
+      return;
+    }
+    // TODO: Fertility Accessory and Characteristic
```

Comment:
I guess this is fine. We should probably add a small note in the README about the fertility being exposed as a light sensor to try and avoid getting issues about it :-)

Reply:
Done 👌